### ISSUE TITLE
glog parser's "glog_timestamp" field can't be used as the event's timestamp

### ISSUE THREAD

**Issue by seh** [honeycombio/honeycomb-kubernetes-agent#35]
For the parsers like "json" and "keyval" that confront timestamp fields as text, it's possible to [use the "timefield" and "drop_field" processors to "move" those fields to the primary timestamp of the posted log event](https://github.com/honeycombio/honeycomb-kubernetes-agent/blob/5258e17a2583720c57d849db1929797aca5cfdb2/processors/timefield.go#L29-L32). That is, say that there's a log record in "keyval" format with a field named "ts," with its values as RFC 3339 strings. You can write the following to make the "ts" field's time the event's time:
```yaml
- dataset: "My Dataset"
  namespace: kube-system
  labelSelector: "app = prometheus-node-exporter"
  parser: keyval
  processors:
  - timefield:
      field: time
      format: "2006-01-02T15:04:05Z07:00" # RFC 3339
  - drop_field:
      field: time
```
However, with the "glog" format, it tries to help by [parsing the expected time string into a `time.Time` value, and stashing it in the "glog_timestamp" field](https://github.com/honeycombio/honeycomb-kubernetes-agent/blob/5258e17a2583720c57d849db1929797aca5cfdb2/parsers/glog.go#L54-L58). It's not possible to later "move" this "glog_timestamp" field into the event's timestamp, because [the "timefield" processor only consumes fields with values of type `string` or `int`](https://github.com/honeycombio/honeycomb-kubernetes-agent/blob/5258e17a2583720c57d849db1929797aca5cfdb2/vendor/github.com/honeycombio/honeytail/httime/httime.go#L105-L113).

It would be nice if the "timefield" processor could treat the "format" configuration field as optional ([maybe it already does](https://github.com/honeycombio/honeycomb-kubernetes-agent/blob/5258e17a2583720c57d849db1929797aca5cfdb2/vendor/github.com/honeycombio/honeytail/httime/httime.go#L172-L196)), and if it's absent, tolerate extracting a field that's already of type `time.Time`. If it could do that, we could promote the "glog_timestamp" field like this:
```yaml
- dataset: "My Dataset"
  namespace: kube-system
  labelSelector: "k8s-app notin (calico-node, calico-typha)"
  parser: glog
  processors:
  - timefield:
      field: glog_timestamp
  - drop_field:
      field: glog_timestamp
```

**Comment by tredman** [comment]
This is fixed in 1.3.0

**Comment by seh** [comment]
I confirmed that this works as expected. Thank you.
